PolloPollo.Repository/ApplicationRepository.cs

```diff
@@ -10,5 +16,140 @@ public ApplicationRepository(PolloPolloContext context)
+        /// </summary>
+        /// <param name="applicationId"></param>
+        /// <returns name="ApplicationDTO"></returns>
+        public async Task<ApplicationDTO> FindAsync(int applicationId)
+        {
+            var application = await (from a in _context.Applications
+                                 where a.Id == applicationId
+                                 select new ApplicationDTO
+                                 {
+                                     ApplicationId = a.Id,
+                                     UserId = a.UserId,
+                                     ProductId = a.ProductId,
+                                     Motivation = a.Motivation,
+                                     TimeStamp = a.TimeStamp,
+                                     Status = a.Status
+                                 }).SingleOrDefaultAsync();
+
+            if (application == null)
+            {
+                return null;
+            }
+
+            return application;
+        }
+
+        /// <summary>
+        /// Retrieve all open applications
+        /// </summary>
+        /// <returns></returns>
+        public IQueryable<ApplicationDTO> Read()
+        {
+            var entities = from a in _context.Applications
+                           where a.Status == ApplicationStatus.Open
+                           select new ApplicationDTO
+                           {
+                               ApplicationId = a.Id,
+                               UserId = a.UserId,
+                               ProductId = a.Id,
+                               Motivation = a.Motivation,
+                               TimeStamp = a.TimeStamp,
+                               Status = a.Status
+                           };
+
+            return entities;
+        }
+
+        /// <summary>
+        /// Retrieve all applications by specified receiver
+        /// </summary>
+        /// <param name="receiverId"></param>
+        /// <returns></returns>
+        public IQueryable<ApplicationDTO> Read(int receiverId)
+        {
+            var entities = from a in _context.Applications
+                           where a.UserId == receiverId
+                           select new ApplicationDTO
+                           {
+                               ApplicationId = a.Id,
+                               UserId = a.UserId,
+                               ProductId = a.ProductId,
+                               Motivation = a.Motivation,
+                               TimeStamp = a.TimeStamp,
+                               Status = a.Status
+                           };
+
+            return entities;
+        }
+
+        /// <summary>
+        /// Delete an application by id
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns name="bool"></returns>
+        public async Task<bool> DeleteAsync(int id)
+        {
+            var application = _context.Applications.Find(id);
```

Comment:
This method should take both the application id and user id.
And if application's user id is not the same as the given user id then return false.